x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/MlAutoscalingExtension.java

```diff
@@ -16,6 +16,10 @@
 public class MlAutoscalingExtension implements AutoscalingExtension {
     private final MachineLearning plugin;
 
+    public MlAutoscalingExtension() {
+        this.plugin = null;
```

Comment:
Please do it like this: https://github.com/elastic/elasticsearch/blob/2b3dbdebf37176c983fccf7bf4325f467b4d1d6c/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/ReservedSecurityStateHandlerProvider.java#L22

Otherwise, if anyone did ever call this constructor they'd be setting up a subsequent null pointer exception.